contracts/staking/TokenStaking.sol

```diff
@@ -1361,15 +1356,34 @@ contract TokenStaking is Ownable, IStaking {
         return tAmountToSlash;
     }
 
-    /// @notice Returns amount of Keep stake in the Keep staking contract for the specified operator.
-    ///         Resulting value in T denomination
-    function getKeepAmountInT(address operator) internal view returns (uint96) {
-        uint256 keepStakeAmount = keepStakingContract.eligibleStake(
-            operator,
-            address(this)
-        );
-        (uint96 tAmount, ) = keepToT(keepStakeAmount);
-        return tAmount;
+    /// @notice Removes application with zero authorization from authorized
+    ///         applications array
+    function cleanAuthorizedApplications(
+        OperatorInfo storage operatorStruct,
+        uint256 numberToDelete
+    ) internal {
+        uint256 length = operatorStruct.authorizedApplications.length;
+        if (numberToDelete == length) {
+            delete operatorStruct.authorizedApplications;
+            return;
+        }
+
+        uint256 deleted = 0;
+        for (
+            uint256 index = 0;
+            index < length - numberToDelete && deleted < numberToDelete;
```

Comment:
Say that `numberToDelete = 3` and there are 6 `operatorStruct.authorizedApplications`: `_ _ _ _ _ _` . 

`length = 6`, `length - numberToDelete = 3`.

`index` will iterate through `[0, 1, 2]`. What if the application we are looking for is at the very end?

Reply:
thanks, during optimization I broke main logic, will fix

Reply:
I hope I fixed it
✔️